lld/ELF/Writer.cpp

```diff
@@ -1543,6 +1543,8 @@ template <class ELFT> void Writer<ELFT>::finalizeAddressDependentContent() {
 
   uint32_t pass = 0, assignPasses = 0;
   for (;;) {
+    if (ctx.arg.relocatable)
```

Comment:
Not calling `createThunks` seems like it might be a pretty big behavior change; AFAICT there aren't hooks to disable it if `relocatable`. This is definitely a gap in my knowledge: is it intended that thunks be created as a result of relocatable linking?

Reply:
Ah, thanks for the explanation. Makes sense to me.

In that case; I think we can remove the check for `ctx.arg.relocatable` in RISC-V's `relaxOnce`. We can now take it as part of the contract for relaxation and thunk creation that they do not occur in relocatable links.

Reply:
Makes sense. created https://github.com/llvm/llvm-project/pull/152240 as a prerequisite